zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobCompleteProcessor.java

```diff
@@ -60,29 +63,48 @@ public void afterAccept(
       final Intent intent,
       final JobRecord value) {
 
-    final var serviceTaskKey = value.getElementInstanceKey();
+    final var elementInstanceKey = value.getElementInstanceKey();
 
-    final ElementInstance serviceTask = elementInstanceState.getInstance(serviceTaskKey);
+    final var elementInstance = elementInstanceState.getInstance(elementInstanceKey);
 
-    if (serviceTask != null) {
-      if (value.getJobKind() == JobKind.EXECUTION_LISTENER) {
-        // to store the variable for merge, to handle concurrent commands
-        eventHandle.triggeringProcessEvent(value);
+    if (elementInstance != null) {
+      switch (value.getJobKind()) {
+        case EXECUTION_LISTENER:
+          {
+            // to store the variable for merge, to handle concurrent commands
+            eventHandle.triggeringProcessEvent(value);
 
-        commandWriter.appendFollowUpCommand(
-            serviceTaskKey,
-            ProcessInstanceIntent.COMPLETE_EXECUTION_LISTENER,
-            serviceTask.getValue());
-        return;
-      }
+            commandWriter.appendFollowUpCommand(
+                elementInstanceKey,
+                ProcessInstanceIntent.COMPLETE_EXECUTION_LISTENER,
+                elementInstance.getValue());
+            return;
+          }
+        case TASK_LISTENER:
+          {
+            // to store the variable for merge, to handle concurrent commands
+            eventHandle.triggeringProcessEvent(value);
 
-      final long scopeKey = serviceTask.getValue().getFlowScopeKey();
-      final ElementInstance scopeInstance = elementInstanceState.getInstance(scopeKey);
+            // TODO explain here why we retrieve intermediate state and not the `regular` user task
```

Comment:
❓ Probably we should resolve this TODO before the PR merge.

Reply:
Thanks Ana for highlighting this!
I've added the explanation comment instead of // TODO in the scope of this [commit](https://github.com/camunda/camunda/pull/22576/commits/65e06b31728360a30fcf5bc68a10bac29483f3fb).